pkg/cover/backend/elf.go

```diff
@@ -417,23 +424,65 @@ func readCoverPoints(file *elf.File, info *symbolInfo) ([2][]uint64, error) {
 	if err != nil {
 		return pcs, fmt.Errorf("failed to read .text: %v", err)
 	}
-	end := len(data) - callLen + 1
+	end := len(data)
+	if arch == targets.AMD64 {
+		end = end - callLen + 1
+	}
 	for i := 0; i < end; i++ {
-		pos := bytes.IndexByte(data[i:end], 0xe8)
+		pos := bytes.IndexByte(data[i:end], opcode)
 		if pos == -1 {
 			break
 		}
 		pos += i
 		i = pos
-		off := uint64(int64(int32(binary.LittleEndian.Uint32(data[pos+1:]))))
 		pc := text.Addr + uint64(pos)
-		target := pc + off + callLen
+		var off uint64
+		var target uint64
+		if arch == targets.AMD64 {
+			off = uint64(int64(int32(binary.LittleEndian.Uint32(data[pos+1:]))))
+			target = pc + off + uint64(callLen)
+		} else if arch == targets.ARM64 {
+			b := data[pos-callLen+1 : pos]
+			b = append(b, 0x00)
```

Comment:
This overwrites the .text data referenced by the elf file because the b slice refers to the same data and capacity is not fixed. This can break calculations in the next loop as we already have 0 bytes in some random locations, and this may break some future code that will assume that .text is unmodified.
One option to fix this would be to cap capacity with:
```
b := data[pos-callLen+1 : pos: callLen-1]
```
but this may lead to memory allocations on append.
So I think the better way to do this is:
```
b := data[pos-callLen+1 : pos+1]
off := uint64(int64(int32(binary.LittleEndian.Uint32(b)))) & (1<<24)-1
```